### PR TITLE
Notifications Implementation

### PR DESCRIPTION
**Pull request by CarsonF**
Following up on #3300 to provide implemented functionality with both databases.

### CODE REVIEW

File: dbschema/notifications.esdl [hunk 1]
```diff
@@ -0,0 +1,34 @@
+module default {
+  abstract type Notification extending Mixin::Audited {
+    readAt := .currentRecipient.readAt; 
+    unread := not exists .currentRecipient.readAt;
+    single currentRecipient := assert_single((
+      select .recipients filter .user = global currentUser
+    ));
+    recipients := .<notification[is Notification::Recipient];
```
[the remark this answers is not in the document]
Comment (CarsonF) [reply dbschema/notifications.esdl:8]: Correct

File: src/components/notifications/notification.edgedb.repository.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      }),
+    );
+
+    const query = e.select({
+      dto: e.select(created, this.hydrate),
+      totalRecipients: e.count(insertedRecipients),
+    });
+
+    const res = await this.db.run(query);
+    return res;
```
[the remark this answers is not in the document]
Comment (CarsonF) [reply src/components/notifications/notification.edgedb.repository.ts:90]: Fair

File: dbschema/notifications.esdl [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  type System extending default::Notification {
+    required message: str;
+  }
+  abstract type Comment extending default::Notification {
+    required comment: Comments::Comment {
+      on target delete delete source;
+    };
+  }
+  type CommentMentioned extending Comment;
```
[the remark this answers is not in the document]
Comment (CarsonF) [reply dbschema/notifications.esdl:32]: True.
That does follow suit with the other one.
What about `MentionedInComment Notification`?

[the remark this answers is not in the document]
Comment (bryanjnelson) [reply dbschema/notifications.esdl:32]: I do like `MentionedInComment` better than `CommentMentioned`.  It bothers me that it's not uniform with other one, but I still prefer it over what is there now.

[the remark this answers is not in the document]
Comment (CarsonF) [reply dbschema/notifications.esdl:32]: How about
```
Notification::CommentViaMention / CommentViaMentionNotification
Notification::CommentViaMembership / CommentViaMembershipNotification
```

[the remark this answers is not in the document]
Comment (rdonigian) [reply dbschema/notifications.esdl:32]: This is definitely a naming improvement over the PR.

[the remark this answers is not in the document]
Comment (bryanjnelson) [reply dbschema/notifications.esdl:32]: YES!

**Pull request merged**